allocator_test.go

```diff
@@ -184,6 +184,74 @@ func TestTagsWithNonviableNodes(t *testing.T) {
 	require.Nil(t, allocation)
 }
 
+func TestMaxChurnWithIncreasedNodeCount(t *testing.T) {
```

Comment:
Can you add a couple more tests with smaller `range + node` sets that are easier to assert on explicitly.